### PR TITLE
feat: initial implementation of ECS Terraform Module

### PR DESCRIPTION
**Pull request by alannix-lw**

### CODE REVIEW

**Comment by afiune** [comment]
Oops, that shouldn't have happened. 🤦🏽‍♂️

File: main.tf [hunk 1]
```diff
@@ -1 +1,193 @@
...
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_task_execution_policy_attachment" {
+  count = var.use_existing_iam_role ? 0 : 1
+
+  role       = aws_iam_role.ecs_execution[0].name
```
Comment (afiune) [inline main.tf]: I think it is better to do this:
```suggestion
  role       = aws_iam_role.ecs_execution[count.index].name
```

File: main.tf [hunk 2]
```diff
@@ -1 +1,193 @@
...
+    }
+  }
+}
+
+resource "aws_iam_policy" "ssm_parameter_store_policy" {
+  count = (!var.use_existing_iam_role && var.use_ssm_parameter_store) ? 1 : 0
+
+  name        = local.iam_role_name
+  description = "An IAM policy to allow a Lacework Agent ECS task to pull the agent access token from SSM"
+  policy      = data.aws_iam_policy_document.ssm_parameter_store_policy[0].json
```
Comment (afiune) [inline main.tf]: same here, what do you think?
```suggestion
  policy      = data.aws_iam_policy_document.ssm_parameter_store_policy[count.index].json
```

File: main.tf [hunk 3]
```diff
@@ -1 +1,193 @@
...
+}
+
+resource "aws_iam_role_policy_attachment" "ssm_parameter_store_iam_role_policy" {
+  count = (!var.use_existing_iam_role && var.use_ssm_parameter_store) ? 1 : 0
+
+  role       = local.iam_role_name
+  policy_arn = aws_iam_policy.ssm_parameter_store_policy[0].arn
```
Comment (afiune) [inline main.tf]: only if you agree 😅:
```suggestion
  policy_arn = aws_iam_policy.ssm_parameter_store_policy[count.index].arn
```

**Review by afiune — CHANGES_REQUESTED** [review]
I feel confused with the use of `use_ssm_parameter_store` and `use_existing_ssm_parameter`. I think we should do:
* Enable `use_ssm_parameter_store` to create the SSM parameter OR
* Let the user pass an existing parameter via `use_existing_ssm_parameter`

But not both.

**Comment by alannix-lw** [comment]
Thanks for the review @afiune - you're the best! Corrected the duplicate effort on having to provide `use_existing_ssm_parameter` and the ARN.

File: README.md [hunk 4]
```diff
@@ -15,21 +15,45 @@ The `main.tf` file will configure an ECS Task Definition which will then be used
...
+| iam_role_name           | The IAM role name to use when `use_existing_iam_role` is `false`         | `string`      | `""`                       |    no    |
+| iam_role_tags           | The tags to apply to a created IAM role                                  | `map(string)` | `{}`                       |    no    |
+| lacework_access_token   | The access token for the Lacework agent                                  | `string`      | n/a                        |   yes    |
+| lacework_task_cpu       | The quantity of CPU units to assign to the task                          | `string`      | `"512"`                    |    no    |
+| lacework_task_mem       | The quantity of Memory (MiB) to assign to the task                       | `string`      | `"512"`                    |    no    |
+| resource_prefix         | A prefix that will be use at the beginning of every generated resource   | `string`      | `"lacework-ecs"`           |    no    |
+| ssm_parameter_arn       | The SSM parameter ARN to use when `use_existing_ssm_parameter` is `true` | `string`      | `""`                       |    no    |
```
Comment (afiune) [inline README.md]: Still see `use_existing_ssm_parameter`

File: examples/existing-ssm-parameter-kms/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+module "lacework_ecs_datacollector" {
+  source = "lacework/ecs-agent/aws"
+  version = "~> 0.1"
+
+  ecs_cluster_arn       = "arn:aws:ecs:us-east-1:123456789012:cluster/example-cluster"
+  lacework_access_token = "0123456789ABCDEF0123456789ABCDEF"
+
+  use_ssm_parameter_store    = true
+  use_existing_ssm_parameter = true
```
Comment (afiune) [inline examples/existing-ssm-parameter-kms/README.md]: Still see `use_existing_ssm_parameter` here

File: examples/existing-ssm-parameter/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+module "lacework_ecs_datacollector" {
+  source = "lacework/ecs-agent/aws"
+  version = "~> 0.1"
+
+  ecs_cluster_arn       = "arn:aws:ecs:us-east-1:123456789012:cluster/example-cluster"
+  lacework_access_token = "0123456789ABCDEF0123456789ABCDEF"
+
+  use_ssm_parameter_store    = true
+  use_existing_ssm_parameter = true
```
Comment (afiune) [inline examples/existing-ssm-parameter/README.md]: Still see `use_existing_ssm_parameter` here

File: variables.tf [hunk 7]
```diff
@@ -1,5 +1,93 @@
...
+variable "resource_prefix" {
+  type        = string
+  default     = "lacework-ecs"
+  description = "A prefix that will be use at the beginning of every generated resource"
+}
+
+variable "ssm_parameter_arn" {
+  type        = string
+  default     = ""
+  description = "The SSM parameter ARN to use when `use_existing_ssm_parameter` is `true`"
```
Comment (afiune) [inline variables.tf]: Still see `use_existing_ssm_parameter` here

**Review by afiune — CHANGES_REQUESTED** [review]
One more change and we are gooooood to GO!

**Review by afiune — APPROVED** [review]
tenor-130743533

**Pull request merged**